File: c++_Programs/fibonacci.cpp
#include <iostream>
#include <vector>

int main() {
    int n = 10;
    std::vector<unsigned long long> f(n);
    if (n > 0) f[0] = 0;
    if (n > 1) f[1] = 1;
    for (int i = 2; i < n; ++i) f[i] = f[i-1] + f[i-2];
    std::cout << "First " << n << " Fibonacci numbers:\n";
    for (auto val : f) std::cout << val << " ";
    std::cout << std::endl;
    return 0;
}

File: Python_Programs/fibonacci.py
def fibonacci(n: int) -> list[int]:
    """Generate first n Fibonacci numbers."""
    if n < 1:
        raise ValueError("N must be >= 1")
    fib = [0, 1]
    for _ in range(2, n):
        fib.append(fib[-1] + fib[-2])
    return fib[:n]

def main():
    try:
        n = int(input("Enter N (>=1): "))
        result = fibonacci(n)
        print(f"Fibonacci sequence: {' '.join(map(str, result))}")
    except ValueError as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()